tests/integration/ha_tests/helpers.py

```diff
@@ -0,0 +1,361 @@
+    """
+    for rel in ops_test.model.relations:
+        endpoints = [endpoint.name for endpoint in rel.endpoints]
+        if endpoint_one in endpoints and endpoint_two in endpoints:
+            return True
+    return False
+
+
+async def get_process_pid(
+    ops_test: OpsTest, unit_name: str, container_name: str, process: str
+) -> int:
+    """Return the pid of a process running in a given unit.
+
+    Args:
+        ops_test: The ops test object passed into every test case
+        unit_name: The name of the unit
+        container_name: The name of the container in the unit
+        process: The process name to search for
+    Returns:
+        A integer for the process id
+    """
+    get_pid_commands = [
+        "ssh",
+        "--container",
+        container_name,
+        unit_name,
+        "pgrep",
+        process,
+    ]
+    return_code, pid, _ = await ops_test.juju(*get_pid_commands)
+
+    assert (
+        return_code == 0
+    ), f"Failed getting pid, unit={unit_name}, container={container_name}, process={process}"
+
+    stripped_pid = pid.strip()
+    if not stripped_pid:
+        return -1
+
+    return int(stripped_pid)
+
+
+async def send_signal_to_pod_container_process(
+    ops_test: OpsTest, unit_name: str, container_name: str, process: str, signal_code: str
+) -> None:
+    """Send the specified signal to a pod container process.
+
+    Args:
+        ops_test: The ops test framework
+        unit_name: The name of the unit to send signal to
+        container_name: The name of the container to send signal to
+        process: The name of the process to send signal to
+        signal_code: The code of the signal to send
+    """
+    kubernetes.config.load_kube_config()
+
+    pod_name = unit_name.replace("/", "-")
+
+    send_signal_command = f"pkill --signal {signal_code} -f {process}"
+    response = kubernetes.stream.stream(
+        kubernetes.client.api.core_v1_api.CoreV1Api().connect_get_namespaced_pod_exec,
+        pod_name,
+        ops_test.model.info.name,
+        container=container_name,
+        command=send_signal_command.split(),
+        stdin=False,
+        stdout=True,
+        stderr=True,
+        tty=False,
+        _preload_content=False,
+    )
+    response.run_forever(timeout=5)
+
+    assert (
+        response.returncode == 0
+    ), f"Failed to send {signal_code} signal, unit={unit_name}, container={container_name}, process={process}"
+
+
+async def get_cluster_status(ops_test: OpsTest, unit: Unit) -> Dict:
+    """Get the cluster status by running the get-cluster-status action.
+
+    Args:
+        ops_test: The ops test framework
+        unit: The unit on which to execute the action on
+
+    Returns:
+        A dictionary representing the cluster status
+    """
+    get_cluster_status_action = await unit.run_action("get-cluster-status")
+    cluster_status_results = await get_cluster_status_action.wait()
+    return cluster_status_results.results
```

Comment:
I don't think this charm has this action. I don't think we should have it. please remove

Reply:
Removed copypasta